packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -188,6 +196,40 @@ export class HardhatNode extends EventEmitter {
     return [common, node];
   }
 
+  private static _validateHardforks(
```

Comment:
We validate this here because:
1. The config validation can't make requests and fetch the remote chainId.
2. The `node` task can provide a different block number so the config validation won't be enough.